pkg/csls/stream.go

```diff
@@ -67,10 +69,16 @@ func (w *Stream) PutCloudfoundryLog(log cloudfoundry.Log, groupName string) erro
 			},
 		},
 	}
-	b, err := json.Marshal(data)
+	json, err := json.Marshal(data)
```

Comment:
although not technically an error, masking the package `json` with the variable `json` can be a bit confusing ... suguest picking a different name.

However you could also rewrite this gzipping process to happen in one go without intermediate files at all... something like:

```
		var b bytes.Buffer
		gz := gzip.NewWriter(&out)
		enc := json.NewEncoder(gz)
		if err := enc.Encode(testContext); err != nil {
			return return fmt.Errorf("failed-to-marshal-batch: %s", err)
		}
```

Reply:
I've fixed the package masking issue. 

Will the compiler not optimise away the intermediate values?

Reply:
No, there will not be any optimising away in the way you have it currently.

You have multiple buffers allocated and the process is happening serially (first the data is encoded as json, THEN it is copied and processed into gzip) .... if you chain the io.Writers as per my example the processes of encoding and gziping can happen concurrently where possible ... and it also avoids the need to manually handle low level `Write()` calls (which isn't fun to handle as you should be checking the number of bytes written and errors) which I notice you are currently not checking the return values of at all

Reply:
Thanks for the streaming example. I'm using that now. I've also added nil checks where necessary.